share/spice/flights/route/flights_route.js

```diff
@@ -209,79 +237,96 @@
-                    minItemsForModeSwitch: 3,
-                    sourceName: 'FlightStats',
-                    sourceUrl: "http://www.flightstats.com/go/FlightStatus/flightStatusByRoute.do?"
-                        + "departure=" + this.sourceCity
-                        + "&arrival=" + this.destinationCity,
-                    itemType: results.length === 1 ? "flight" : "flights by departure time" 
-                },
-                normalize: function(item) {
-                
-                    return {
-                        url: "http://www.flightstats.com/go/FlightStatus/flightStatusByFlight.do?"
-                            + "airlineCode=" + item.airlineCode 
-                            + "&flightNumber=" + item.flightNumber
-                            + "&departureDate=" 
-                            + item.departureDate.getFullYear() + "-"
-                            + (item.departureDate.getMonth() + 1) + "-"
-                            + item.departureDate.getDate()
-                    }
-                },
-                sort_fields: {
-                    departureDate: function(a, b) {
-                        a = +new Date(a.departureDate);
-                        b = +new Date(b.departureDate);
-                        return ((a < b) ? -1 : (a > b ? 1 : 0));
-                    }
-                },
-                sort_default: 'departureDate',
-                templates: {
-                    group: 'base',
-                    detail: false,
-                    options: {
-                        content: Spice.flights_route.content
+            DDG.require('moment.js', function() {
+                Spice.add({
+                    data: results,
+                    id: "route",
+                    name: "Route",
+                    meta: {
+                        minItemsForModeSwitch: 3,
+                        sourceName: 'FlightStats',
+                        sourceUrl: "http://www.flightstats.com/go/FlightStatus/flightStatusByRoute.do?"
+                            + "departure=" + this.sourceCity
+                            + "&arrival=" + this.destinationCity,
+                        itemType: 'flights'
+                    },
+                    normalize: function(item) {
+                        var status_text,
+                            arrival_datetime;
+
+                        if (!item.is_on_time) {
+                            status_text = 'Scheduled';
+                            arrival_datetime = moment(item.scheduledArrivalDate);
+                        } else {
+                            if (moment(item.arrivalDate).isBefore(moment())) {
+                                status_text = 'Arrived';
+                                arrival_datetime = moment(item.arrivalDate);
+                            } else {
+                                status_text = 'Arrives';
+                                arrival_datetime = moment(item.scheduledArrivalDate);
```

Comment:
Looks like there's a bug here? I'm seeing a landed flight that says "Arrives in 9 hours"

![qatar_airways_doha_to_islamabad_at_duckduckgo](https://cloud.githubusercontent.com/assets/873785/9557682/e7bdd63a-4dab-11e5-8c53-d24806dcb60e.png)